spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/MongoMetricsAutoConfigurationTests.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.assertj.AssertableApplicationContext;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link MongoMetricsAutoConfiguration}.
+ *
+ * @author Chris Bono
+ */
+class MongoMetricsAutoConfigurationTests {
+
+	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(MongoMetricsAutoConfiguration.class));
+
+	@Test
+	void whenThereIsAMeterRegistryThenMetricsListenersAreAdded() {
+		this.contextRunner.with(MetricsRun.simple())
+				.withConfiguration(AutoConfigurations.of(MongoAutoConfiguration.class)).run((context) -> {
+					assertThat(context).hasSingleBean(MongoMetricsCommandListener.class);
+					assertThat(getActualMongoClientSettingsUsedToConstructClient(context)).isNotNull()
+							.extracting(MongoClientSettings::getCommandListeners).asList()
+							.containsExactly(context.getBean(MongoMetricsCommandListener.class));
+				});
+	}
+
+	@Test
+	void whenThereIsNoMeterRegistryThenNoMetricsListenersAreAdded() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(MongoAutoConfiguration.class))
+				.run(this::assertThatMetricsListenerNotAdded);
+	}
+
+	@Test
+	void whenThereIsNoMongoClientSettingsOnClasspathThenNoMetricsListenersAreAdded() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(MongoAutoConfiguration.class))
+				.withClassLoader(new FilteredClassLoader(MongoClientSettings.class))
+				.run(this::assertThatMetricsListenerNotAdded);
+	}
+
+	@Test
+	void whenThereIsNoMongoMetricsCommandListenerOnClasspathThenNoMetricsListenersAreAdded() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(MongoAutoConfiguration.class))
+				.withClassLoader(new FilteredClassLoader(MongoMetricsCommandListener.class))
+				.run(this::assertThatMetricsListenerNotAdded);
+
+	}
+
+	@Test
+	void whenMetricsCommandListenerEnabledPropertyFalseThenNoMetricsListenersAreAdded() {
+		this.contextRunner.withConfiguration(AutoConfigurations.of(MongoAutoConfiguration.class))
+				.withPropertyValues("management.metrics.mongo.command-listener.enabled:false")
+				.run(this::assertThatMetricsListenerNotAdded);
+	}
+
+	private void assertThatMetricsListenerNotAdded(final AssertableApplicationContext context) {
+		assertThat(context).doesNotHaveBean(MongoMetricsCommandListener.class);
+		assertThat(getActualMongoClientSettingsUsedToConstructClient(context)).isNotNull()
+				.extracting(MongoClientSettings::getCommandListeners).asList().isEmpty();
+	}
+
+	private MongoClientSettings getActualMongoClientSettingsUsedToConstructClient(
+			final AssertableApplicationContext context) {
+		final MongoClient mongoClient = context.getBean(MongoClient.class);
```

Comment:
Sad but true, we need to get at the actually used [MongoClientSettings](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mongo/MongoAutoConfiguration.java#L54) where the builder customizer is applied. Its not exposed so we reflect on it.